src/test/scala/io/picnicml/doddlemodel/preprocessing/RangeScalerTest.scala

```diff
@@ -0,0 +1,31 @@
+package io.picnicml.doddlemodel.preprocessing
+
+import breeze.linalg.DenseMatrix
+import io.picnicml.doddlemodel.TestingUtils
+import io.picnicml.doddlemodel.data.Feature.{CategoricalFeature, FeatureIndex, NumericalFeature}
+import org.scalatest.{FlatSpec, Matchers}
+import io.picnicml.doddlemodel.preprocessing.RangeScaler.ev
+import org.scalactic.{Equality, TolerantNumerics}
+
+class RangeScalerTest extends FlatSpec with Matchers with TestingUtils {
+  implicit val doubleTolerance: Equality[Double] = TolerantNumerics.tolerantDoubleEquality(1e-4)
+
+  "Range scaler" should "scale features to specified range" in {
```

Comment:
Once the preprocessor is implemented in a way so that all columns are retained in the output, we would need two different tests; one that transforms all numerical columns (`RangeScaler((0.0, 1.0), featureIndex)`) and one with a subset of numerical columns (`RangeScaler((0.0, 1.0), featureIndex)`). Take a look at the standard scaler tests for examples.